packages/runtime/container-runtime/src/channelCollection.ts

```diff
@@ -1148,6 +1152,36 @@ export class ChannelCollection implements IFluidDataStoreChannel, IDisposable {
 		return builder.getSummaryTree();
 	}
 
+	/**
+	 * Gets the GC data. Used when attaching or serializing a detached container.
+	 */
+	public getAttachGCData(telemetryContext?: ITelemetryContext): IGarbageCollectionData {
+		const builder = new GCDataBuilder();
+		// Attaching graph of some stores can cause other stores to get bound too.
+		// So keep taking summary until no new stores get bound.
+		let notBoundContextsLength: number;
+		do {
+			notBoundContextsLength = this.contexts.notBoundLength();
+			// Iterate over each data store and ask for its GC data.
+			Array.from(this.contexts)
+				.filter(
+					([key, _]) =>
+						// Take GC data of bounded data stores only.
+						!this.contexts.isNotBound(key),
+				)
+				.map(([key, value]) => {
+					const contextGCData = value.getAttachGCData(telemetryContext);
+					// Prefix the child's id to the ids of its GC nodes so they can be identified as belonging to the child.
+					// This also gradually builds the id of each node to be a path from the root.
+					builder.prefixAndAddNodes(key, contextGCData.gcNodes);
+				});
+		} while (notBoundContextsLength !== this.contexts.notBoundLength());
+
+		// Get the outbound routes (aliased data stores) and add a GC node for this channel.
+		builder.addNode("/", Array.from(this.aliasedDataStores));
```

Comment:
While you're here, do we have a way to get the absolute path of this channel collection?  For nested data stores, it won't be "/".

Fine to track as a follow-up, if someone would just tell us where!

Reply:
The best way would be to create a handle for this channel and use it's path. Agreed on a follow-up.